112.md

```diff
@@ -0,0 +1,248 @@
+NIP-112
+=======
+
+Encrypted Group Events
+----------------------
+
+`draft` `optional` `author:earonesty` 
+
+This NIP defines new event kinds for encrypted group events, with chat as the primary example, and with optional forward secrecy guidelines. 
+
+This NIP builds on the encryption from NIP-44, the gift-wrap of NIP-59 and the distributed chat channel management of NIP-28.   It is similar to NIP38, with the exception that gift-wrap is used on every message, and issues like metadata leakage and optional forward secrecy are addressed.   
+
+It works by creating a shared secret that is used to encrypt the group or channel messages. Only the participants know that secret, so only such members can read/write to this group chat.  This effectively hides metadata from external users.   See drawbacks below.
+
+It reserves 8 event kinds (1059, 400-406) for immediate use.
+
+- [`1059`](https://github.com/nostr-protocol/nips/pull/468) - encrypted gift-wrap, can be used by many protocols
+- `400`  - wrapped kind: create encrypted channel
+- `401`  - wrapped kind: invite to encrypted channel
+- `402`  - wrapped kind: change channel metadata
+- `403`  - wrapped kind: send encrypted chat message
+- `404`  - wrapped kind: moved to new channel
+- `405`  - wrapped kind: delegate new owner
+- `406`  - wrapped kind: accept owner delegation
+
+
+## Kind 1059: Encrypted Gift Wrap
+
+For all encrypted group chat messages, the client first generates a kind 4XX message, as below.   
+
+Then they generate a new public/private keypair, and uses this to sign and encrypt the message.
+
+The `content` is [NIP-44](https://github.com/nostr-protocol/nips/pull/574) encrypted JSON string with the temporary key, this is the inner event.
+
+All events called "wrapped-kind XX" refer to these kind 1059 gift wrapped events.
```

Comment:
```suggestion
For all encrypted group chat messages, the client first generates a kind 400 message, using a fresh public/private keypair.

The `content` is [NIP-44](https://github.com/nostr-protocol/nips/pull/574) encrypted JSON string with the temporary key, this is the inner event.

All events called "wrapped-kind XX" refer to kind XX events wrapped using kind 1059 gift wrapped events.
```

Reply:
this is not correct.   the "wrapped kind 4XX" messages are always using the same keypair.   only the kind 1059 is fresh.   i will make this more clear.